### ISSUE TITLE
derive_hash_xor_eq not issued for struct with lifetime parameter

### ISSUE THREAD

**Issue by magicant** [rust-lang/rust-clippy#8810]
### Summary

When I manually implement PartialEq and derive Hash for a single struct, I expect a derive_hash_xor_eq warning. However, if the struct has a lifetime parameter, clippy does not seem to issue this warning.

### Lint Name

derive_hash_xor_eq

### Reproducer

I tried this code:

```rust
#[derive(Debug, Eq, Hash)]
struct Foo<'a>(&'a str);

impl<'a> PartialEq for Foo<'a> {
    fn eq(&self, other: &Foo<'a>) -> bool {
        self.0 == other.0
    }
}

fn main() {
    assert_ne!(Foo("foo"), Foo("bar"));
}
```

I expected to see this happen: The code is rejected by the default-enabled lint `derive_hash_xor_eq`.

Instead, this happened: No warnings.

### Version

```text
rustc 1.60.0 (7737e0b5c 2022-04-04)
binary: rustc
commit-hash: 7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c
commit-date: 2022-04-04
host: x86_64-unknown-linux-gnu
release: 1.60.0
LLVM version: 14.0.0
```

**Comment by lukaslueg** [comment]
Current stable [lints](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=0307e3e28ac1f3d997ad9414bd857dae) (notice that the lint has been renamed). Same as #12092 ?!

**Comment by magicant** [comment]
This seems to have been fixed indeed. Thank you!
